flare/ase/calculator.py

```diff
@@ -158,9 +163,11 @@ def build_mgp(self, skip=True):
         grid_params['svd_rank_2'] = rank_2
         grid_params['svd_rank_3'] = rank_3
        
-        hyps = self.gp_model.hyps
-        cutoffs = self.gp_model.cutoffs
-        self.mgp_model = MappedGaussianProcess(hyps, cutoffs,
-                        grid_params, struc_params, mean_only,
-                        container_only, self.gp_model, lmp_file_name)
-
+        self.mgp_model = MappedGaussianProcess(grid_params,
+                                               struc_params,
+                                               map_force=map_force,
+                                               GP=self.gp_model,
+                                               mean_only=mean_only,
+                                               container_only=container_only,
+                                               lmp_file_name=lmp_file_name,
+                                               n_cpus=n_cpus)
```

Comment:
I'm in general a huge fan of passing in as much as possible via keyword arguments this way for non user-facing functions; helps to make sure that the function is robust against later re-factoring which changes the names of the arguments.

Reply:
As we discussed, this part need re-considering. I might just change it to a "rebuild" method